pkg/irreparable/irreparable.go

```diff
@@ -0,0 +1,150 @@
+// Copyright (C) 2018 Storj Labs, Inc.
+// See LICENSE for copying information.
+
+package irreparable
+
+import (
+	"context"
+	"sync"
+	"time"
+
+	"go.uber.org/zap"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+	monkit "gopkg.in/spacemonkeygo/monkit.v2"
+
+	"storj.io/storj/internal/migrate"
+	dbx "storj.io/storj/pkg/irreparable/dbx"
+)
+
+var (
+	mon = monkit.Package()
+)
+
+// Database implements the irreparable RPC service
+type Database struct {
+	mu     sync.Mutex
+	DB     *dbx.DB
+	logger *zap.Logger
+}
+
+// RemoteSegmentInfo is info about a single entry stored in the irreparable db
+type RemoteSegmentInfo struct {
+	SegmentPath        []byte
+	SegmentDetail      []byte
+	LostPiecesCount    int64
+	RepairUnixSec      int64
+	RepairAttemptCount int64
+}
+
+// NewServer creates instance of Server
+func NewServer(source string, logger *zap.Logger) (*Database, error) {
+	db, err := dbx.Open("sqlite3", source)
+	if err != nil {
+		return nil, err
+	}
+
+	err = migrate.Create("irreparable", db)
+	if err != nil {
+		return nil, err
+	}
+
+	return &Database{
+		DB:     db,
+		logger: logger,
+	}, nil
+}
+
+func (db *Database) locked() func() {
+	db.mu.Lock()
+	return db.mu.Unlock
+}
+
+// Put a db entry for the provided remote segment info
+func (db *Database) Put(ctx context.Context, rmtSegInfo *RemoteSegmentInfo) (err error) {
+	defer mon.Task()(&ctx)(&err)
+	// entry exists already?
+	updateResp, err := db.Get(ctx, rmtSegInfo)
+	if err != nil {
+		return db.Create(ctx, rmtSegInfo)
+	}
+	return db.Update(ctx, updateResp)
+}
+
+// Create a db entry for the provided remote segment info
+func (db *Database) Create(ctx context.Context, info *RemoteSegmentInfo) (err error) {
```

Comment:
Instead of having all of `Put`, `Create`, `Update`... let's just have one method `IncrementRepairAttempts(ctx, segmentInfo)`

Reply:
Done